controller.go

```diff
@@ -206,6 +196,66 @@ func New(cfgOptions ...config.Option) (NetworkController, error) {
 	return c, nil
 }
 
+func (c *controller) ReloadConfiguration(cfgOptions ...config.Option) error {
```

Comment:
Should we protect this function with the buffered channel to make sure only one caller is active at a given time ?